modules/compiler/utils/source/work_item_loops_pass.cpp

```diff
@@ -1611,7 +1645,14 @@ Function *compiler::utils::WorkItemLoopsPass::makeWrapperFunction(
         }
       }
 
+      const auto &successors = barrierMain.getSuccessorIds(i);
+      const auto num_succ = successors.size();
+      const bool barrierInTrapBB = isBarrierInTrapBB(barrierMain, i);
+
       auto *const exitBlock = [&]() {
+        if (num_succ == 0 && barrierInTrapBB) {
+          return schedule.makeRunOneWorkItem(block, i, true);
+        }
```

Comment:
maybe an assert if we get into this position with no barrierInTrapBB since this means something has gone wrong.

Reply:
Yeah we should hit the `llvm_unreachable` [here](https://github.com/PietroGhg/oneapi-construction-kit/blob/111af696544a721ca1cb07e42586f6b5012fca30/modules/compiler/utils/source/work_item_loops_pass.cpp#L1698) if `num_succ == 0` without `barrierInTrapBB`, but I can add an assert here too